backend/packages/Upgrade/src/api/controllers/ExperimentClientController.v5.ts

```diff
@@ -289,15 +289,22 @@ export class ExperimentClientController {
     // append userDoc in logger
     request.logger.child({ userDoc: experimentUserDoc });
     request.logger.info({ message: 'Got the original user doc' });
-    const { id, group } = await this.experimentUserService.updateGroupMembership(
-      experimentUser.id,
+    const udateResult = await this.experimentUserService.updateGroupMembership(
+      experimentUserDoc.requestedUserId,
       experimentUser.group,
       {
         logger: request.logger,
         userDoc: experimentUserDoc,
       }
     );
-    return { id, group };
+    if (!udateResult) {
+      request.logger.error({
+        details: 'update unexpectedly returned empty object',
+      });
+      throw new InternalServerError('set group membership failed');
+    }
+
+    return { id: experimentUserDoc.id, group: experimentUser.workingGroup };
   }
```

Comment:
Shouldn't the `setGroupMemberShip` function return `experimentUser.group` instead of `experimentUser.workingGroup`?